src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/IHandlerContext.kt

```diff
@@ -69,8 +71,9 @@ interface IHandlerContext {
      * Sends an [event]
      *
      * @param event event to send
+     * @param rootEventId optional rootEventId different from session rootEventId
      */
-    fun send(event: Event)
+    fun send(event: Event, rootEventId: EventID? = null): EventID
```

Comment:
Maybe we should add @JvmOverloads annotation here (otherwise, this is not a compatible change for java)